### PR TITLE
TurntoBall working and bugs resolved in geometry and math_functions

### PR DESCRIPTION
**Pull request by RKJenamani**
Commits are made from the main system in the lab.

### CODE REVIEW

File: utils/geometry.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def intersects(self,center,radius):
 			return False
 	# self is the center of circle, checks whether the line made by the point1 and point2 intersects the circle
 	def intersects(self,point1,point2,radius):
-		# Source of algorithm used: http://stackoverflow.com/questions/1073336/circle-line-collision-detection 
+		# Source of algorithm used: http://stackoverflow.com/questions/10733336/circle-line-collision-detection
```
Comment (saurabhkgp21) [inline utils/geometry.py]: @RKJenamani why change this link?

Comment (RKJenamani) [reply utils/geometry.py]: Someone must have done it by mistake. Did not notice.

Comment (saurabhkgp21) [reply utils/geometry.py]: change this to previous one

**Comment by saurabhkgp21** [comment]
Do not update gotoball if itsn't working good. Just test it properly and update it later

**Comment by RKJenamani** [comment]
Yeah, sorry I just wanted to commit to my fork for later use.
I'll revert them.

**Pull request merged**